### ISSUE TITLE
fatal error: 'ET.pb.h' file not found

### ISSUE THREAD

**Issue by yurivict** [MisterTea/EternalTerminal#528]
Build fails on some FreeBSD systems with such error:
```
===>  Building for eternalterminal-6.2.1
[  1% 1/82] cd /wrkdirs/usr/ports/net/eternalterminal/work/.build && /usr/local/bin/cmake -DCOTIRE_BUILD_TYPE:STRING=Release -P /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/external_imported/cotire/CMake/cotire.cmake combine /wrkdirs/usr/ports/net/eternalterminal/work/.build/et-test_CXX_cotire.cmake /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.cxx /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/src/base/Headers.hpp
[  2% 2/82] cd /wrkdirs/usr/ports/net/eternalterminal/work/.build && /usr/local/bin/cmake -DCOTIRE_BUILD_TYPE:STRING=Release -P /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/external_imported/cotire/CMake/cotire.cmake combine /wrkdirs/usr/ports/net/eternalterminal/work/.build/et-test_CXX_cotire.cmake /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.cxx
[  3% 3/82] cd /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1 && /usr/local/bin/cmake -DCOTIRE_BUILD_TYPE:STRING=Release -P /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/external_imported/cotire/CMake/cotire.cmake precompile /wrkdirs/usr/ports/net/eternalterminal/work/.build/et-test_CXX_cotire.cmake /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx.pch /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/test/BackedTest.cpp
FAILED: cotire/et-test_CXX_prefix.hxx.pch /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx.pch 
cd /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1 && /usr/local/bin/cmake -DCOTIRE_BUILD_TYPE:STRING=Release -P /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/external_imported/cotire/CMake/cotire.cmake precompile /wrkdirs/usr/ports/net/eternalterminal/work/.build/et-test_CXX_cotire.cmake /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx.pch /wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/test/BackedTest.cpp
In file included from /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.hxx:4:
In file included from /wrkdirs/usr/ports/net/eternalterminal/work/.build/cotire/et-test_CXX_prefix.cxx:4:
/wrkdirs/usr/ports/net/eternalterminal/work/EternalTerminal-et-v6.2.1/src/base/Headers.hpp:118:10: fatal error: 'ET.pb.h' file not found
#include "ET.pb.h"
         ^~~~~~~~~
1 error generated.
```

```ET.pb.h``` isn't generated.

Total number of commands in both successful and unsuccessful logs is 82, which suggests that this is a dependency problem.

Full log is  [here](http://beefy16.nyi.freebsd.org/data/130amd64-default/c9a6195b6bd1/logs/errors/eternalterminal-6.2.1.log) (IPv6 URL).

Version: 6.2.0
FreeBSD 13.0

**Comment by yurivict** [comment]
> The link to the full log doesn't work.

Unfortunately, this is an IPv6 URL.

> What happens if you build with a single job?

I wasn't able to reproduce the problem personally. It only occurred on central build servers,
Changing to serial build was my first guess for a workaround. This is what I did. Will see in the next several days if it would fail again.

But you need to check if user of ```T.pb.h``` declared it as dependency.

**Comment by jwmcglynn** [comment]
I mirrored the logs here: https://gist.github.com/jwmcglynn/7bd1654d0d420702a782c1c87d2271c1

Could it be differences due to a different version of make?  I don't recognize that log style.  `ET.pb.h` is one of the generated files for `ET.proto`, so I'm guessing an include path issue?

**Comment by yurivict** [comment]
It uses ninja.

> I'm guessing an include path issue?

No. In successful logs ET.pb.h appears in the first commands. In failed logs it doesn't.
I am guessing that it was scheduled to be rebuilt later, after it is used.

**Comment by yurivict** [comment]
The serial build also has this problem on that specific system.

**Comment by MisterTea** [comment]
https://github.com/MisterTea/EternalTerminal/blob/master/CMakeLists.txt#L391

**Comment by yurivict** [comment]
When I look at ```Makefile``` that cmake generates I don't see that ```et-lib``` depends on ```generated-code``` at all.

**Comment by yurivict** [comment]
I created this cmake [bug report](https://gitlab.kitware.com/cmake/cmake/-/issues/23762).

Please also note that you use the feature ```add_dependencies``` that appeared in cmake-3.3, but you require cmake version 3.0.2, which is a contradiction.

I think that this triggers a buggy behavior in cmake. It seems to ignore this function completely.

**Comment by MisterTea** [comment]
If you require cmake 3.3 instead, does the problem go away.

**Comment by yurivict** [comment]
> If you require cmake 3.3 instead, does the problem go away?

Makefile doesn't change. So it probably doesn't go away. I can't myself reproduce the actual problem since it only occurs on build servers.

Since cmake closed the ticket without explanation I opened this [Discord forum discussion](https://discourse.cmake.org/t/why-add-dependencies-doesnt-have-any-effect/6131).

**Comment by nunotexbsd** [comment]
@yurivict 

It seems that build problem is fixed in 6.2.8.
I pushed it to ports in:
https://cgit.freebsd.org/ports/commit/?id=0d5cdf4e4c3c96c0defe60e73bc9c19fd965781d
https://cgit.freebsd.org/ports/commit/?id=1cf2efcfbb7d9376221c073ad0abd88c91644895

I'm testing on 2 machines client/server:

- Server seg faults with signal 6 when a nmap is done to it
- Both client and server logs appear splited in /tmp (default in etc config) one for each connection
- Connection is made ok (if nothing interferes)
```
etclient-2024-02-12_16-08-08.log
etclient-2024-02-12_16-08-32.log

etclient-stderr-2024-02-12_16-08-08.log
etclient-stderr-2024-02-12_16-08-32.log

etserver-2024-02-12_16-07-04_1644.log
etserver-2024-02-12_16-08-26_1694.log

etserver-stderr-2024-02-12_16-07-04_1644.log
etserver-stderr-2024-02-12_16-08-26_1694.log

etterminal-nunotex-7RsXRyAtVtpv1RA3-2024-02-12_16-07-29.log
etterminal-nunotex-QpH8NFn6gx9D0XU2-2024-02-12_16-10-20.log
```

I was expecting same log to include all connections in:
etclient.log and etserver.log...
Maybe I missing some better config.

Could you do some testing?

Cheers

**Comment by MisterTea** [comment]
That looks correct. I'm in the middle of moving the system tests to GitHub actions. For now, the tests you ran are good enough.
